sources/System/Animator/AnimatorSystem.cpp

```diff
@@ -0,0 +1,47 @@
+//
+// Created by hbenjamin on 03/06/2021.
+//
+
+#include <Component/Animator/AnimatorComponent.hpp>
+#include <Component/Keyboard/KeyboardComponent.hpp>
+#include <Controllers/Keyboard.hpp>
+#include <Model/Model.hpp>
+#include <Component/Animation/AnimationsComponent.hpp>
+#include <Component/Controllable/ControllableComponent.hpp>
+#include "AnimatorSystem.hpp"
+#include "Component/Renderer/Drawable3DComponent.hpp"
+
+using Keyboard = RAY::Controller::Keyboard;
+namespace RAY3D = RAY::Drawables::Drawables3D;
+using Key = RAY::Controller::Keyboard::Key;
+
+namespace BBM
+{
+	AnimatorSystem::AnimatorSystem()
+			: WAL::System({
+				typeid(AnimatorComponent),
+				typeid(ControllableComponent)
+			})
+	{}
+
+	void AnimatorSystem::onFixedUpdate(WAL::Entity &entity)
+	{
+		if (!entity.hasComponent<ControllableComponent>())
+			return;
+		const std::vector<std::vector<float>> moveDiag = {{-1, 0}, {-1, 1}, {0, 1}, {1, 1}, {1, 0}, {1, -1}, {0, -1}, {-1, -1}};
```

Comment:
Instead of doing it manually, use the Vector3.angle() (use the velocity vector inside the MovableComponent)